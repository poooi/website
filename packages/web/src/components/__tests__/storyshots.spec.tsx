import initStoryshots from '@storybook/addon-storyshots'
import { render } from '@testing-library/react'
// eslint-disable-next-line import/no-extraneous-dependencies
import type { Plugin } from 'pretty-format'

const reactTestingLibrarySerializer: Plugin = {
  print: (val: any, serialize) =>
    serialize(val.getByTestId('storybook-content')),
  test: (val: any) =>
    val && Object.prototype.hasOwnProperty.call(val, 'container'),
}

initStoryshots({
  renderer: render,
  snapshotSerializers: [reactTestingLibrarySerializer],
})
