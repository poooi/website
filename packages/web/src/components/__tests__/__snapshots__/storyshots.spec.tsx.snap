// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Animated Container Normal 1`] = `
<div
  data-testid="storybook-content"
>
  <div>
    Hello world
  </div>
</div>
`;

exports[`Storyshots Background Dark 1`] = `
.c0 {
  z-index: -1;
  position: fixed;
  top: 0;
  left: 0;
  background-color: #2F343C;
}

<div
  data-testid="storybook-content"
>
  <canvas
    class="c0"
    height="768"
    style="width: 1024px; height: 768px;"
    width="1024"
  />
</div>
`;

exports[`Storyshots Background Light 1`] = `
.c0 {
  z-index: -1;
  position: fixed;
  top: 0;
  left: 0;
  background-color: #F6F7F9;
}

<div
  data-testid="storybook-content"
>
  <canvas
    class="c0"
    height="768"
    style="width: 1024px; height: 768px;"
    width="1024"
  />
</div>
`;

exports[`Storyshots Download Beta Version Available 1`] = `
.c0 {
  text-align: center;
  min-height: 150px;
}

.c0 .ms-Spinner-label {
  font-size: 20px;
}

.c3 {
  display: block;
  font-size: 1rem;
}

.c2 {
  padding: 1rem 2em;
  font-size: 1.5rem;
}

.c2.c2 {
  background-color: #2F343C;
  color: #D3D8DE;
  background-image: none;
  border: 1px solid #D3D8DE;
}

.c1 + .c1 {
  margin-left: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem auto;
}

<div
  data-testid="storybook-content"
>
  <div
    class="c0"
  >
    <a
      class="bp4-button bp4-intent-primary c1 c2"
      data-testid="download-stable-version"
      href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
      role="button"
      tabindex="0"
    >
      <span
        class="bp4-button-text"
      >
        <span>
          Download v10.2.4
        </span>
        <span
          class="c3"
        >
          Windows (x64)
        </span>
        <span
          class="c3"
        >
          Stable, for most users
        </span>
      </span>
    </a>
    <a
      class="bp4-button c1 c2"
      href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-setup-11.0.0-beta.0.exe"
      role="button"
      tabindex="0"
    >
      <span
        class="bp4-button-text"
      >
        <div>
          Download v11.0.0-beta.0
        </div>
        <span
          class="c3"
        >
          Windows (x64)
        </span>
        <span
          class="c3"
        >
          Beta, for new features
        </span>
      </span>
    </a>
  </div>
  <div
    class="c4"
  >
    <button
      class="bp4-button bp4-minimal"
      data-testid="open-dialog"
      href="/downloads"
      type="button"
    >
      <span
        class="bp4-button-text"
      >
        Choose another platform
      </span>
    </button>
  </div>
</div>
`;

exports[`Storyshots Download Stable Version Only 1`] = `
.c0 {
  text-align: center;
  min-height: 150px;
}

.c0 .ms-Spinner-label {
  font-size: 20px;
}

.c3 {
  display: block;
  font-size: 1rem;
}

.c2 {
  padding: 1rem 2em;
  font-size: 1.5rem;
}

.c2.c2 {
  background-color: #2F343C;
  color: #D3D8DE;
  background-image: none;
  border: 1px solid #D3D8DE;
}

.c1 + .c1 {
  margin-left: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem auto;
}

<div
  data-testid="storybook-content"
>
  <div
    class="c0"
  >
    <a
      class="bp4-button bp4-intent-primary c1 c2"
      data-testid="download-stable-version"
      href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
      role="button"
      tabindex="0"
    >
      <span
        class="bp4-button-text"
      >
        <span>
          Download v10.2.4
        </span>
        <span
          class="c3"
        >
          Windows (x64)
        </span>
        <span
          class="c3"
        >
          Stable, for most users
        </span>
      </span>
    </a>
  </div>
  <div
    class="c4"
  >
    <button
      class="bp4-button bp4-minimal"
      data-testid="open-dialog"
      href="/downloads"
      type="button"
    >
      <span
        class="bp4-button-text"
      >
        Choose another platform
      </span>
    </button>
  </div>
</div>
`;

exports[`Storyshots Error Boundary Throws 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.2em;
}

.c1 {
  margin-bottom: 1em;
}

<div
  data-testid="storybook-content"
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Ah, the webpage has become üê¢. We will fix it.
    </div>
    <div>
      <button
        type="button"
      >
        Reload
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Target List Beta Version Available 1`] = `
.c0 {
  text-align: center;
  min-height: 150px;
}

.c0 .ms-Spinner-label {
  font-size: 20px;
}

.c3 {
  display: block;
  font-size: 1rem;
}

.c2 {
  padding: 1rem 2em;
  font-size: 1.5rem;
}

.c2.c2 {
  background-color: #2F343C;
  color: #D3D8DE;
  background-image: none;
  border: 1px solid #D3D8DE;
}

.c1 + .c1 {
  margin-left: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 button,
.c4 a {
  margin: 1ex 1ex;
}

<div
  data-testid="storybook-content"
>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <div
      class="c0"
    >
      <a
        class="bp4-button bp4-intent-primary c1 c2"
        data-testid="download-stable-version"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          <span>
            Download v10.2.4
          </span>
          <span
            class="c3"
          >
            Windows (x64)
          </span>
          <span
            class="c3"
          >
            Stable, for most users
          </span>
        </span>
      </a>
      <a
        class="bp4-button c1 c2"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-setup-11.0.0-beta.0.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          <div>
            Download v11.0.0-beta.0
          </div>
          <span
            class="c3"
          >
            Windows (x64)
          </span>
          <span
            class="c3"
          >
            Beta, for new features
          </span>
        </span>
      </a>
    </div>
  </div>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <h2>
      Stable
       
      v10.2.4
    </h2>
    <div
      class="c4"
    >
      <a
        class="bp4-button bp4-intent-success"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Linux (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-arm64.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Linux (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi_10.2.4_amd64.deb"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Debian / Ubuntu (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi_10.2.4_arm64.deb"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Debian / Ubuntu (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4.x86_64.rpm"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Fedora / CentOS (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4.dmg"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          macOS (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-arm64.dmg"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          macOS (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x64) Portable
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x86)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-ia32-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x86) Portable
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-arm64-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (ARM)
        </span>
      </a>
    </div>
  </div>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <h2>
      Beta
       
      v11.0.0-beta.0
    </h2>
    <div
      class="c4"
    >
      <a
        class="bp4-button bp4-intent-success"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-setup-11.0.0-beta.0.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Linux (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0-arm64.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Linux (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi_11.0.0-beta.0_amd64.deb"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Debian / Ubuntu (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi_11.0.0-beta.0_arm64.deb"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Debian / Ubuntu (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0.x86_64.rpm"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Fedora / CentOS (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0.dmg"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          macOS (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0-arm64.dmg"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          macOS (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x64) Portable
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-setup-11.0.0-beta.0.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x86)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0-ia32-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x86) Portable
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v11.0.0-beta.0/poi-11.0.0-beta.0-arm64-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (ARM)
        </span>
      </a>
    </div>
  </div>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <h2>
      Others
    </h2>
    <div
      class="c4"
    >
      <a
        class="bp4-button"
        data-testid="old-versions"
        href="https://registry.npmmirror.com/binary.html?path=poi/"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Old versions
          <br />
          Hosted at TAONPM
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://ci.appveyor.com/project/KochiyaOcean/poi"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Windows nightlies
          <br />
          Hosted at AppVeyor
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://nightly.poi.moe/"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Linux and macOS nightlies
          <br />
          Hosted at poi.moe
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://github.com/poooi/poi"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Source code
          <br />
          Hosted at GitHub
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Storyshots Target List Stable Version Only 1`] = `
.c0 {
  text-align: center;
  min-height: 150px;
}

.c0 .ms-Spinner-label {
  font-size: 20px;
}

.c3 {
  display: block;
  font-size: 1rem;
}

.c2 {
  padding: 1rem 2em;
  font-size: 1.5rem;
}

.c2.c2 {
  background-color: #2F343C;
  color: #D3D8DE;
  background-image: none;
  border: 1px solid #D3D8DE;
}

.c1 + .c1 {
  margin-left: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 button,
.c4 a {
  margin: 1ex 1ex;
}

<div
  data-testid="storybook-content"
>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <div
      class="c0"
    >
      <a
        class="bp4-button bp4-intent-primary c1 c2"
        data-testid="download-stable-version"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          <span>
            Download v10.2.4
          </span>
          <span
            class="c3"
          >
            Windows (x64)
          </span>
          <span
            class="c3"
          >
            Stable, for most users
          </span>
        </span>
      </a>
    </div>
  </div>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <h2>
      Stable
       
      v10.2.4
    </h2>
    <div
      class="c4"
    >
      <a
        class="bp4-button bp4-intent-success"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Linux (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-arm64.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Linux (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi_10.2.4_amd64.deb"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Debian / Ubuntu (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi_10.2.4_arm64.deb"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Debian / Ubuntu (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4.x86_64.rpm"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Fedora / CentOS (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4.dmg"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          macOS (x64)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-arm64.dmg"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          macOS (ARM)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x64) Portable
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-setup-10.2.4.exe"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x86)
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-ia32-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (x86) Portable
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://npmmirror.com/mirrors/poi/v10.2.4/poi-10.2.4-arm64-win.7z"
        role="button"
        tabindex="0"
      >
        <span
          class="bp4-button-text"
        >
          Windows (ARM)
        </span>
      </a>
    </div>
  </div>
  <div
    class="bp4-card bp4-elevation-0"
  >
    <h2>
      Others
    </h2>
    <div
      class="c4"
    >
      <a
        class="bp4-button"
        data-testid="old-versions"
        href="https://registry.npmmirror.com/binary.html?path=poi/"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Old versions
          <br />
          Hosted at TAONPM
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://ci.appveyor.com/project/KochiyaOcean/poi"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Windows nightlies
          <br />
          Hosted at AppVeyor
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://nightly.poi.moe/"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Linux and macOS nightlies
          <br />
          Hosted at poi.moe
        </span>
      </a>
      <a
        class="bp4-button"
        href="https://github.com/poooi/poi"
        rel="noopener noreferrer"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp4-button-text"
        >
          Source code
          <br />
          Hosted at GitHub
        </span>
      </a>
    </div>
  </div>
</div>
`;
